tt_metal/impl/device/device.cpp

```diff
@@ -3706,6 +3742,55 @@ const std::vector<SubDeviceId> &Device::get_sub_device_ids() const {
     return this->active_sub_device_manager_->get_sub_device_ids();
 }
 
+std::vector<CoreCoord> Device::get_optimal_dram_bank_to_logical_worker_assignment() {
```

Comment:
from 10K feet, not clear why this code is here and the other "get_optimal" are in core_assignment.  Also, fn comment would be helpful (how do these routines work together)

Reply:
This function accesses physical coordinates, which are restricted to live in the `Device` class and shouldn't leak into other parts of the code base.
It derives physical coordinates and then passes them off to the functions in `core_assignment.cpp` which use this info for actually assigning cores to banks.
I've added a comment describing the distribution of work across functions.